api/v1alpha1/externaldns_types.go

```diff
@@ -507,6 +510,13 @@ type ExternalDNSServiceSourceOptions struct {
 	ServiceType []corev1.ServiceType `json:"serviceType,omitempty"`
 }
 
+type ExternalDNSServiceOpenShiftRouteOptions struct {
+	// If source is openshift-route then you can pass the ingress controller name or names. Based on this name external-dns will select the respective router.
+	// +kubebuilder:validation:Required
+	// +required
+	OCPRouterName string `json:"ocpRouterName"`
```

Comment:
```suggestion
	RouterName string `json:"routerName"`
```
AFAIK OCP is a RH product but this operator should work in OKD as well.